### PR TITLE
ccls: init at 0.20181225.7

### PR DESCRIPTION
**Pull request by Mic92**
- tested with [emacs](https://dl.thalheim.io/kdh-PwxzlwGKTEl1_NpTzg/2019-01-13-190156_1920x1080_scrot.png) and vim on nix and systemd.
- wrapped to pick up our cc wrapper environment -> works perfectly in nix-shell

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by Mic92** [comment]
cc @tobim 
@GrahamcOfBorg build ccls

**Comment by Mic92** [comment]
cc @dtzWill

**Comment by Mic92** [comment]
Looks like the release was deleted -> opened an upstream ticket: https://github.com/MaskRay/ccls/issues/202

File: pkgs/development/tools/misc/ccls/default.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
+{ stdenv, fetchFromGitHub, makeWrapper
+, cmake, llvmPackages, rapidjson }:
+
+stdenv.mkDerivation rec {
+  name    = "ccls-${version}";
+  version = "0.20181225.6";
```
Comment (MaskRay) [inline pkgs/development/tools/misc/ccls/default.nix]: Sorry but if it didn't bother you much, could you please use the tag 0.20181225.7 ?

0.20181225.6 may cause unpleasant log lines in vscode and coc.nvim. https://github.com/MaskRay/ccls/issues/205

File: pkgs/development/tools/misc/ccls/default.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+    mv $out/bin/ccls $out/bin/$wrapped
+    substituteAll ${./wrapper} $out/bin/ccls
+    chmod --reference=$out/bin/$wrapped $out/bin/ccls
+  '';
+
+  meta = with stdenv.lib; {
+    description = "A c/c++ language server powered by libclang";
+    homepage    = https://github.com/MaskRay/ccls;
+    license     = licenses.mit;
```
Comment (MaskRay) [inline pkgs/development/tools/misc/ccls/default.nix]: It is Apache 2. Every file has been rewritten.

File: pkgs/development/tools/misc/ccls/default.nix [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    wrapped=".ccls-wrapped"
+    export wrapped
+
+    mv $out/bin/ccls $out/bin/$wrapped
+    substituteAll ${./wrapper} $out/bin/ccls
+    chmod --reference=$out/bin/$wrapped $out/bin/ccls
+  '';
+
+  meta = with stdenv.lib; {
+    description = "A c/c++ language server powered by libclang";
```
[the remark this answers is not in the document]
Comment (Mic92) [reply pkgs/development/tools/misc/ccls/default.nix]: Too much copy-pasta from cquery :)

**Comment by Mic92** [comment]
@GrahamcOfBorg build ccls

**Pull request merged**